Modules/_functoolsmodule.c

```diff
@@ -1423,6 +1423,11 @@ static PyType_Spec lru_cache_type_spec = {
     .slots = lru_cache_type_slots
 };
 
+/* helper function to make typing.NewType.__call__ method faster */
+
+PyObject* functools_idfunc(PyObject* self, PyObject* x) {
+    return x;
```

Comment:
Does this need to INCREF x?

Reply:
It does, fixed. Thanks for review

Reply:
Sorry I don't understand why this needs an incref.

If lets say you do  ``x = NewType('blah', int)``, then ``int`` would already be auto incref-ed by ``STORE_X`` instruction right?

Reply:
I will try to make things clear:

```py
from typing import NewType

X = NewType("X", int)  # here we do not call _functools._idfunc

n = X(1)  # here we have call to _functools._idfunc
```

Reply:
Woops I forgot how `__call__` worked for a moment so I gave a wrong example. Sorry!